packages/ui/src/components/vuestic-components/va-toast/index.ts

```diff
@@ -0,0 +1,164 @@
+        const value = source[prop as OptionKeys]
+        if (value !== undefined) {
+          target[prop] = value
+        }
+      }
+    }
+  })
+  return target
+}
+
+
+const initNotification = (options: NotificationOptions | string) => {
+  const toastInstance: VaToast = createToastInstance(getToastOptions(options))
+  toastInstances.push(toastInstance)
+  return toastInstance
+}
+
+const closeNotification = (id: any) => {
+  if (!toastInstances.length) {
+    seed = 1
+    return
+  }
+  const closableInstance = toastInstances.find((toastInstance: any) => toastInstance.id === id)
+  if (!closableInstance) {
+    return
+  }
+
+  const closableInstanceIndex = toastInstances.findIndex((toastInstance: any) => toastInstance.id === id)
+
+  // if (typeof closableInstance.onClose() === 'function') {
+  //   closableInstance.onClose()
+  // }
+
+  const closableInstancePosition = closableInstance.position
+  const removedHeight = toastInstance.$el.offsetHeight
+
+  toastInstances = toastInstances.reduce((acc: any[], toastInstance: any, index: number) => {
+    if (index === closableInstanceIndex) {
+      toastInstance.$el.style.visibility = 'hidden'
+      return acc
+    }
+    //Check for following instance and modify it position if true
+    const isFollowingInstance: boolean = index > closableInstanceIndex && toastInstance.position === closableInstancePosition
+    if (isFollowingInstance) {
+      toastInstance.$el.style[toastInstance.positionY] =
+        parseInt(toastInstance.$el.style[toastInstance.positionY], 10) - removedHeight - 16 + 'px'
+      acc.push(toastInstance)
+      return acc
+    }
+    return [...acc, toastInstance]
+  }, [])
+  if (!toastInstances.length) seed = 1
+}
+
+const closeAllNotifications = () => {
+  if (!toastInstances.length) {
+    seed = 1
+    return
+  }
+
+  toastInstances.forEach((toastInstance: any) => {
+    closeNotification(toastInstance.id)
+  })
+}
+
+const getToastOptions = (options: string | NotificationOptions): any => {
+  if (typeof options === 'string') {
+    options = {
+      message: options,
+    }
+  }
+
+  const onCloseHandler = options.onClose
+  const id: string = 'notification_' + seed
+  options.onClose = () => {
+    if (onCloseHandler) {
+      onCloseHandler()
+    }
+    Notification.close(id, onCloseHandler)
+  }
+  return merge({}, options)
+}
+
+const createToastInstance = (options: NotificationOptions): VaToast => {
+  const id: string = 'notification_' + seed++
+  toastInstance = new NotificationConstructor({
+    propsData: options,
+  })
+
+  const position: string = toastInstance.position
+
+  if (isVNode(options.message)) {
+    toastInstance.$slots.default = [options.message as VNode]
+    options.message = 'REPLACED_BY_VNODE'
+  }
+  toastInstance.id = id
+  toastInstance.$mount()
+  document.body.appendChild(toastInstance.$el)
+  toastInstance.visible = true
+  ;(toastInstance.$el as HTMLElement).style.zIndex = Z_INDEX + ''
```

Comment:
semicolon?

Reply:
it's ts hack. Without it, TS try to merge 125 and 126 lines and try to call **true(toastInstance.$el as HTMLElement)**